share/wake/lib/versions/v0_24.wake

```diff
@@ -1,7 +1,7 @@
isNE isNone isPass isPathDir isSome jempty jfilter jfind jlist job jsonEscape lastValueInList lcm len listToTree listToTreeMulti listToVector localRunner logDebug logEcho logError logInfo logNever logReport logWarning machine makeError makeExecPlan makeJSONRunner makeJSONRunnerPlan makeLogLevel makeLogLevel2 makePlan makeRunner makeShellPlan map mapFlat mapPartial matches max mergeJSON min mkdir mkdirIn nan normalizeJSON normalizeJSONCompat normalizeJSONIdentity normalizeJSONWith ofilter omap omapPartial or orElse overrideJSON parseJSONBody parseJSONFile path pi powm prepend prependPlanPath prettyJSON prettyJSONFormat print printLevel println printlnLevel prod quote rOK read regExpCat regExpToString relative replace reverse rfoldl rfoldr rmap rmapFail rmapPass root runIncrementalJob runJob runJobWith scanl scanr scmp scmpCanonical scmpIdentifier scmpLowercase seq setEnvironment setErrorCause setErrorStack setJSONFormatDouble setJSONFormatIndent setJSONFormatInteger setJSONFormatString setJSONRunnerPlanEstimate setJSONRunnerPlanExtraArgs setJSONRunnerPlanExtraEnv setJSONRunnerPlanRawScript setJSONRunnerPlanScore setJobTag setPairFirst setPairSecond setPlanCommand setPlanDirectory setPlanEcho setPlanEnvVar setPlanEnvironment setPlanFnInputs setPlanFnOutputs setPlanKeep setPlanLabel setPlanLocalOnly setPlanOnce setPlanPersistence setPlanResources setPlanRunnerFilter setPlanShare setPlanStderr setPlanStdin setPlanStdout setPlanUsage setPlanVisible setRunnerInputCommand setRunnerInputDirectory setRunnerInputEnvironment setRunnerInputLabel setRunnerInputPrefix setRunnerInputRecord setRunnerInputResources setRunnerInputStdin setRunnerInputVisible setRunnerName setRunnerOutputInputs setRunnerOutputOutputs setRunnerOutputUsage setRunnerScore setTripleFirst setTripleSecond setTripleThird setUsageCPUtime setUsageInBytes setUsageMemBytes setUsageOutBytes setUsageRuntime setUsageStatus shellJob simplify sortBy sortPaths sortStrings source sources splitAt splitBy splitUntil sqrt stack str strHex strOctal strbase stringToRegExp strlen sum sysname tab tail take takeUntil tap tappi tat tcontains tdelete tdistinctBy tdistinctRunBy tdrop tdropUntil tempty tequal texists tfilter tfind tfold tfoldl tfoldr tforall tinsert tinsertMulti tinsertReplace tintersect tlen tlowerGE tlowerGT tmax tmin tnew tokenize transpose treeToList treeToVector tsplit tsplitAt tsplitBy tsplitUntil tsubset tsubtract ttake ttakeUntil tunion tunionMulti tupperLE tupperLT unicodeCanonical unicodeIdentifier unicodeLowercase unicodeToInteger unreachable unsetEnvironment unzip vat vcmp vdistinctBy vdistinctRunBy vdrop vdropUntil vectorToList vectorToTree vectorToTreeMulti vempty version vexists vfilter vfind vflatten vfoldl vfoldr vforall virtualRunner vlen vmap vmapPartial vmapPartial2 vmapReduce vmapScan vreverse vscan vscanl vscanr vseq vsortBy vsplitAt vsplitBy vsplitUntil vtab vtake vtakeUntil vtranspose vunfoldl vunzip vzip wOK wait wakePath which whichIn workspace write writeIn xOK xor zip
```

Comment:
killJob isn't used anywhere and I'd like to get rid of it. A dummy killJob is worse than just removing it.

Reply:
Maybe a dummy `killJob` that also prints out a deprecation/removal notice to `logError`?  (Since it only returns `Unit`, it's not an interface people could ever explicitly rely on having succeeded.)  I agree it's something we should get rid of, but the versioned packages are meant to be frozen API, and I don't like setting that precedent here.

Reply:
In practical terms for this specific change, nothing.  In the abstract, though, these packages are meant to be a frozen API and removing the function breaks that contract.

Not only is your scenario unlikely, it's also unrealistic.  In order to find `killJob` in the first place in order to use it, someone would have to either be writing for 0.24, 0.27, or 0.28 compatibility and scroll through an overlong, alphabetized list to find a name with no documentation; or be writing for 0.29 compatibility, look at the versioned package source instead of the current stdlib, and ignore the fact that the function they've found very obviously doesn't have any logic filled in.  In both situations, they'd then have to write *new* code using it and ignore the fact that whatever testing they do shows no effect beyond printing an error message explicitly telling them not to use it.  Yeah, someone *might* see only the name and think of a situation they *might* want to use it in, and then be a bit disappointed when it turns out to no longer be supported, but that's no different than any other library which leaves deprecated API in for a few releases.